lib/web/app/handler.go

```diff
@@ -205,3 +206,18 @@ func extractAppName(r *http.Request) (string, error) {
 
 	return parts[0], nil
 }
+
+// ResolveFQDN returns FQDN of the application based on proxy parameters
+func ResolveFQDN(proxyName string, proxyHost string, appClusterName string, app services.App) string {
```

Comment:
please add some unit tests for this

Reply:
agree but this is not a final PR, this is mostly for RJ to proceed with his work. I suggest we do a proper review at the end when it's all ready so we do not do it twice.

Reply:
(as the logic for this method could change)